### PR TITLE
API: Closes #7879: (drops not nan in panel.to_frame() by default)

### LINKED ISSUE #7879: panel.to_frame() discards nan by default
As already described in issue #3690, panel.to_frame discards all minor entries with nan in the data, which can be very confusing. I believe there should be a warning, first time data is dropped or the opposite should be the default behavior.

The warning could be treated similar to a ZeroDivisionWarning in numpy only on the first occurrence.

See below for an example:

``` python
df1 = pd.DataFrame(np.random.randn(2, 3), columns=['A', 'B', 'C'],
                   index=['foo', 'bar'])
df2 = pd.DataFrame(np.random.randn(2, 3), columns=['A', 'B', 'C'],
                   index=['foo', 'bar'])
df2.loc['foo', 'B'] = np.nan
mydict = {'df1': df1, 'df2': df2}

pd.Panel(mydict).to_frame()
```

Output:  

| major | minor | df1 | df2 |
| --- | --- | --- | --- |
| foo | A | 1.9097545931480682 | -0.6710202447941566 |
| foo | C | 1.3335254610685865 | 1.53372538551507 |
| bar | A | 0.3145550744497975 | -1.7221352144306152 |
| bar | B | -0.15681197178861878 | -1.2308510354641322 |
| bar | C | -0.09598971674309852 | -0.1268630728124487 |

Using filter_observations=False, nan won't be dropped:

``` python
pd.Panel(mydict).to_frame(filter_observations=False)
```

Output:  

| major | minor | df1 | df2 |
| --- | --- | --- | --- |
| foo | A | 1.9097545931480682 | -0.6710202447941566 |
| foo | B | 2.092552358833253 |  |
| foo | C | 1.333525461068586 | 1.53372538551507 |
| bar | A | 0.3145550744497975 | -1.7221352144306152 |
| bar | B | -0.15681197178861878 | -1.2308510354641322 |
| bar | C | -0.09598971674309852 | -0.1268630728124487 |

### PR DESCRIPTION
**Pull request by hensing**
closes #7879 by changing API (filter_observations=True now deprecated and replaced with dropna=False)

### CODE REVIEW

File: pandas/core/panel.py [hunk 1]
```diff
@@ -867,7 +868,7 @@ def to_frame(self, filter_observations=True):
 
         Parameters
         ----------
-        filter_observations : boolean, default True
```
Comment (jreback) [inline pandas/core/panel.py]: leave this in (and say its deprecated)

File: pandas/core/panel.py [hunk 2]
```diff
@@ -877,7 +878,7 @@ def to_frame(self, filter_observations=True):
         """
         _, N, K = self.shape
 
-        if filter_observations:
```
Comment (jreback) [inline pandas/core/panel.py]: you need to handle the old value as well (e.g. translate to ``.dropna``)

**Comment by jreback** [comment]
this will need a sub-section in whatsnew/0.17.0. showing what the original behavior is and what the new is.

**Comment by jreback** [comment]
can you update according to comments

File: pandas/core/panel.py (hunk 1 above)
Comment (hensing) [inline pandas/core/panel.py]: ok - will add it

File: pandas/core/panel.py (hunk 2 above)
Comment (hensing) [inline pandas/core/panel.py]: Tested it: it's already being handled correctly by the decorator:
Both, filter_observations and dropna map correctly to the desired behavior.
Hence IMO I don't need to handle it.

**Comment by hensing** [comment]
@jreback: updated the commit according to your comments!

**Comment by jreback** [comment]
- need a test for the deprecation itself.
- need a separate sub-section in the whatsnew explaining this change

**Comment by hensing** [comment]
added/fixed tests and added sub-section

File: doc/source/release.rst [hunk 3]
```diff
@@ -63,6 +63,7 @@ Highlights include:
 - Development support for benchmarking with the `Air Speed Velocity library <https://github.com/spacetelescope/asv/>`_ (:issue:`8316`)
 - Support for reading SAS xport files, see :ref:`here <whatsnew_0170.enhancements.sas_xport>`
 - Removal of the automatic TimeSeries broadcasting, deprecated since 0.8.0, see :ref:`here <whatsnew_0170.prior_deprecations>`
+- Deprecated ``filter_observations`` by ``dropna`` in ``Panel.to_frame`` and changed default to ``True`` (:issue:`7879`)
```
Comment (jreback) [inline doc/source/release.rst]: move this to the whatsnew/v0.17.0 in th edeprecation section. This needs an example in a sub-section. Showing the previous usage and the new.

File: pandas/core/panel.py [hunk 4]
```diff
@@ -870,7 +871,11 @@ def to_frame(self, filter_observations=True):
 
         Parameters
         ----------
-        filter_observations : boolean, default True
+        dropna : boolean, default False
+            Drop (major, minor) pairs without a complete set of observations
+            across all the items
+
+        filter_observations : boolean, default False, [deprecated]
```
Comment (jreback) [inline pandas/core/panel.py]: the default for the original needs to remain

File: pandas/tests/test_panel.py [hunk 5]
```diff
@@ -1525,19 +1525,19 @@ def test_transpose_copy(self):
 
     def test_to_frame(self):
         # filtered
-        filtered = self.panel.to_frame()
+        filtered = self.panel.to_frame(dropna=True)
```
Comment (jreback) [inline pandas/tests/test_panel.py]: need a test for ``.to_frame()``, which is equiv to ``.to_frame(dropna=False)``

File: pandas/core/panel.py (hunk 4 above)
Comment (hensing) [inline pandas/core/panel.py]: How is that possible? If you don't serve the "filter_observations" argument then you don't use the deprecated argument and use dropna instead. With that filter_observations is False implicitly.

File: doc/source/release.rst (hunk 3 above)
Comment (hensing) [inline doc/source/release.rst]: Will do

**Comment by jreback** [comment]
this is pushed to 0.18.0 pls update when you can

**Comment by jreback** [comment]
closing, but if you'd like to update, pls reopen

**Pull request closed without merging**